### DISCUSSION TITLE
[spatie-laravel-translatable-plugin] Builders/Repeaters don't preserve order after language switch and save.

### DISCUSSION THREAD

**Discussion by mybouhssina** [filamentphp/filament#11713] [Help]
### Package

Other

### Package Version

3.2.44

### How can we help you?

Hello,

I might be doing something wrong here, but I've noticed that when I'm using Builders/Repeaters with spatie-laravel-translatable-plugin, if you switch language and submit the form then refresh the page, you will notice that the order of the previous local's Builders/Repeaters has changed  (sometimes). I think it's because javascript doesn't preserve objects order, but once you remove the keys of the builder/repeater elements, it works fine. This might be the reason why both Builder and Repeater are dehydrating their states using array_values.

As a temporary fix, I'm itterrating over my builders/repeaters and calling array_values on them in Trait@updatedActiveLocale:

```php
use Translatable {
        updatedActiveLocale as translatableUpdateActiveLocale;
    }
public function updatedActiveLocale(): void
    {
        $this->translatableUpdateActiveLocale();
        if(isset($this->otherLocaleData[$this->oldActiveLocale]['blocks'])) {
            $this->otherLocaleData[$this->oldActiveLocale]['blocks'] = array_values($this->otherLocaleData[$this->oldActiveLocale]['blocks']);
        }
//...
        $this->form->callAfterStateHydrated(); // check this: https://github.com/filamentphp/filament/discussions/11703
}
//...
```

I'm not sure what a proper solution to this would be... maybe somehow trigger the mutateDehydratedState callback to get the data of a component instead of directly accessing it ?

Thank you.

**Comment by godlikeAlex** [comment]
Wow, man thank you!! YOU SAVE MY LIFE :), maybe you will create an issue? it really interferes with the work when the order changes
